File: telegram-bot/src/main/java/ru/gazprombank/payhub/telegrambot/controller/TelegramBot.java
package ru.gazprombank.payhub.telegrambot.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.bots.TelegramLongPollingBot;
import org.telegram.telegrambots.meta.api.objects.Update;

import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
import ru.gazprombank.payhub.telegrambot.handler.ResponseHandler;

@Slf4j
@Component
@RequiredArgsConstructor
public class TelegramBot extends TelegramLongPollingBot {
    private final ResponseHandler responseHandler;
    @Value(value = "${telegram.bot.name}")
    private String botName;
    @Value(value = "${telegram.bot.token}")
    private String botToken;

    @Override
    public String getBotUsername() {
        return botName;
    }

    @Override
    public String getBotToken() {
        return botToken;
    }

    @Override
    public void onUpdateReceived(Update update) {
        log.info("Update с сообщением {}", update.getMessage().getText());
        try {
            execute(responseHandler.messageHandle(update.getMessage()));
        } catch (TelegramApiException e) {
            log.error("Ошибка отправки {}", e.getMessage());
        }
    }
}

File: telegram-bot/src/main/kotlin/ru/gazprombank/payhub/telegrambot/controller/TelegramBot.kt
package ru.gazprombank.payhub.telegrambot.controller

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import org.telegram.telegrambots.extensions.bots.commandbot.TelegramLongPollingCommandBot
import org.telegram.telegrambots.extensions.bots.commandbot.commands.BotCommand
import org.telegram.telegrambots.meta.api.objects.Update

import ru.gazprombank.payhub.telegrambot.util.createMessage

@Component
class TelegramBot(
    commands: Set<BotCommand>,
    @Value("\${telegram.bot.token}")
    token: String,
) : TelegramLongPollingCommandBot(token) {

    @Value("\${telegram.bot.name}")
    private val botName: String = ""

    init {
        registerAll(*commands.toTypedArray())
    }

    override fun getBotUsername(): String = botName

    override fun processNonCommandUpdate(update: Update) {
        if (update.hasMessage()) {
            val chatId = update.message.chatId.toString()
            if (update.message.hasText()) {
                execute(createMessage(chatId, "Текст пока не обрабатывается"))
            } else {
                execute(createMessage(chatId, "Я понимаю только команды!"))
            }
        }
    }
}